numpyro/infer/util.py

```diff
@@ -360,15 +386,23 @@ def initialize_model(rng_key, model,
     model_kwargs = {} if model_kwargs is None else model_kwargs
     substituted_model = substitute(seed(model, rng_key if jnp.ndim(rng_key) == 1 else rng_key[0]),
                                    substitute_fn=init_strategy)
-    inv_transforms, replay_model, model_trace = _get_model_transforms(
+    inv_transforms, replay_model, has_enumerate_support, model_trace = _get_model_transforms(
         substituted_model, model_args, model_kwargs)
     constrained_values = {k: v['value'] for k, v in model_trace.items()
                           if v['type'] == 'sample' and not v['is_observed']
                           and not v['fn'].is_discrete}
 
+    if has_enumerate_support:
+        from numpyro.contrib.funsor import config_enumerate, enum
+
+        if not isinstance(model, enum):
```

Comment:
If the model is already wrapped by `enum`, we might skip this step... (this check might be unnecessary).